### PR TITLE
dev list-contact

### PR DESCRIPTION
**Pull request by lisperata**
image
image
image
image
image
image
image

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		"await-promise": true,
+		"curly": true,
+		"prefer-for-of": true,
+		"forin": true,
+		"no-console": [
+			true,
+			"log",
+			"error"
+		],
+		"no-debugger": true,
```
Comment (CatDidenko) [inline tslint.json]: Where is formatting? 😀

File: src/list/list.model.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { prop } from '@typegoose/typegoose';
+import { TimeStamps } from '@typegoose/typegoose/lib/defaultClasses';
+import { ObjectId } from 'mongodb';
+import * as MUUID from 'uuid-mongodb';
+import { IListModel } from './list.interfaces';
```
Comment (CatDidenko) [inline src/list/list.model.ts:6]: Please, add blank line after node_modules definitions and before files.

File: src/list/list.controller.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  constructor(private readonly listService: ListService) {}
+  @Post('list')
+  async addList(@Body() dto: CreateListDto): Promise<DocumentType<ListModel>> {
+    return this.listService.create(dto);
+  }
+
+  @Get('lists')
+  async getAllLists(): Promise<DocumentType<ListModel>[]> {
+    return this.listService.getAll();
+  }
```
Comment (CatDidenko) [inline src/list/list.controller.ts:31]: Please, add blank line after }

File: src/contact/dto/update-contact.dto.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { IsString, IsUUID } from 'class-validator';
+
+export class UpdateContactDto {
+  @IsUUID()
+  _id: string;
```
Comment (CatDidenko) [inline src/contact/dto/update-contact.dto.ts:5]: Add blank line after each property definition

File: src/contact/dto/create-contact.dto.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { IsString } from 'class-validator';
+
+export class CreateContactDto {
+  @IsString()
+  name: string;
```
Comment (CatDidenko) [inline src/contact/dto/create-contact.dto.ts:5]: Add blank line after each property definition

File: src/contact/contact.service.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  public async patch(
+    dto: UpdateContactDto,
+  ): Promise<DocumentType<ContactModel> | null> {
+    const { _id, name, email } = dto;
+
+    const newContact: NewContactInPatchType = {};
+    name ? (newContact.name = name) : '';
+    email ? (newContact.email = email) : '';
```
Comment (CatDidenko) [inline src/contact/contact.service.ts]: Strange constructions

File: src/configs/mongo.config.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  'mongodb://' +
+  configService.get('MONGO_LOGIN') +
+  ':' +
+  configService.get('MONGO_PASSWORD') +
+  '@' +
+  configService.get('MONGO_HOST') +
+  ':' +
+  configService.get('MONGO_PORT') +
+  '/' +
+  configService.get('MONGO_AUTHDB');
```
Comment (CatDidenko) [inline src/configs/mongo.config.ts]: Hmm, please, use magic strings https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Template_literals

File: src/app.module.ts [hunk 8]
```diff
@@ -1,9 +1,19 @@
...
+import { TypegooseModule } from 'nestjs-typegoose';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
+import { getMongoConfig } from './configs/mongo.config';
+import { ContactModule } from './contact/contact.module';
+import { ListModule } from './list/list.module';
 
 @Module({
-  imports: [],
+  imports: [ConfigModule.forRoot(),
```
Comment (CatDidenko) [inline src/app.module.ts]: Please, after [ start from new line
Each array item should be on new line
[ 
  ContactModule, 
  ListModule,
]

File: src/list/list.module.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  imports: [
+    TypegooseModule.forFeature([
+      {
+        typegooseClass: ListModel,
+        schemaOptions: {
+          collection: 'List',
+        },
+      },
+    ]),
+  ],
```
Comment (CatDidenko) [inline src/list/list.module.ts]: You don't need this

Comment (CatDidenko) [reply src/list/list.module.ts]: Снимок экрана 2021-07-14 в 19 56 50

export const yoursModel = getModelForClass(YoursModel);

In the module file:
Снимок экрана 2021-07-14 в 19 58 02

**Review by CatDidenko — CHANGES_REQUESTED** [review]

File: src/app.module.ts (hunk 8 above)
Comment (lisperata) [reply src/app.module.ts]: the blank line was added

File: src/configs/mongo.config.ts (hunk 7 above)
Comment (lisperata) [reply src/configs/mongo.config.ts]: Magic strings were used

File: src/contact/contact.service.ts (hunk 6 above)
Comment (lisperata) [reply src/contact/contact.service.ts]: The construction was changed

File: src/contact/dto/create-contact.dto.ts (hunk 5 above)
Comment (lisperata) [reply src/contact/dto/create-contact.dto.ts:5]: the blank lines were added

File: src/contact/dto/update-contact.dto.ts (hunk 4 above)
Comment (lisperata) [reply src/contact/dto/update-contact.dto.ts:5]: the blank line was added

File: src/list/list.controller.ts (hunk 3 above)
Comment (lisperata) [reply src/list/list.controller.ts:31]: the blank line was added

File: src/list/list.model.ts (hunk 2 above)
Comment (lisperata) [reply src/list/list.model.ts:6]: the blank line was added

File: tslint.json (hunk 1 above)
Comment (lisperata) [reply tslint.json]: file was formatted

**Pull request closed without merging**